aws_lambda_builders/workflows/rust_cargo/actions.py

```diff
@@ -0,0 +1,176 @@
+    """
+    Wrapper around file system functions, to make it easy to
+    unit test actions in memory
+    """
+
+    def popen(self, command, stdout=None, stderr=None, env=None, cwd=None):
+        return subprocess.Popen(command, stdout=stdout, stderr=stderr, env=env, cwd=cwd)
+
+    def copyfile(self, source, destination):
+        shutil.copyfile(source, destination)
+
+    def makedirs(self, path):
+        if not os.path.exists(path):
+            os.makedirs(path)
+
+
+class RustBuildAction(BaseAction):
+    NAME = "CargoLambdaBuild"
+    DESCRIPTION = "Building the project using Cargo Lambda"
+    PURPOSE = Purpose.COMPILE_SOURCE
+
+    def __init__(self, source_dir, binaries, mode, architecture=X86_64, handler=None, flags=None, osutils=OSUtils()):
+        """
+        Build the a Rust executable
+
+        :type source_dir: str
+        :param source_dir:
+            Path to a folder containing the source code
+
+        :type binaries: dict
+        :param binaries:
+            Resolved path dependencies
+
+        :type mode: str
+        :param mode:
+            Mode the build should produce
+
+        :type architecture: str, optional
+        :param architecture:
+            Target architecture to build the binary, either arm64 or x86_64
+
+        :type handler: str, optional
+        :param handler:
+            Handler name in `package.bin_name` or `bin_name` format
+
+        :type flags: list, optional
+        :param flags:
+            Extra list of flags to pass to `cargo lambda build`
+
+        :type osutils: object, optional
+        :param osutils:
+            Optional, External IO utils
+        """
+        self._source_dir = source_dir
+        self._mode = mode
+        self._binaries = binaries
+        self._handler = handler
+        self._flags = flags
+        self._architecture = architecture
+        self._osutils = osutils
+
+    def build_command(self):
+        cmd = [self._binaries["cargo"].binary_path, "lambda", "build"]
+        if self._mode == BuildMode.RELEASE:
+            cmd.append("--release")
+        if self._architecture == ARM64:
+            cmd.append("--arm64")
+        if self._handler:
+            cmd.extend(["--bin", self._handler])
+        if self._flags:
+            cmd.extend(self._flags)
+
+        return cmd
+
+    def execute(self):
+        try:
+            command = self.build_command()
+            LOG.debug("Executing cargo-lambda: %s", " ".join(command))
+            if LOG.isEnabledFor(logging.DEBUG):
+                os.environ["RUST_LOG"] = "debug"
+
+            p = self._osutils.popen(
+                command,
+                stderr=subprocess.PIPE,
+                stdout=subprocess.PIPE,
+                cwd=self._source_dir,
+            )
+            out, err = p.communicate()
+            if p.returncode != 0:
+                raise RustBuilderError(message=err.decode("utf8").strip())
+            return out.decode("utf8").strip()
```

Comment:
I think it would be nice to print `out` (and `err` if it is not empty) through logging. I saw you set debug environment variable, which is great to investigate what is going on with building here, but I assume we need to print process outputs as well so that user can see what is executed and what was the result there.

nit: and while doing that, it would be better if we can rename the `p` to something more verbose (`rust_process` or `cargo_process`)